src/crypto/index.js

```diff
@@ -65,6 +65,32 @@ export function isCryptoAvailable() {
     return Boolean(global.Olm);
 }
 
+/* subscribes to timeline events for SAS verification */
+function listenForEvents(client, roomId, listener) {
+    let isEncrypted = false;
+    if (roomId) {
+        isEncrypted = client.isRoomEncrypted(roomId);
+    }
+
+    if (isEncrypted) {
+        client.on("Event.decrypted", listener);
+    } else {
+        client.on("event", listener);
```

Comment:
I think we need to listen on `event` even in encrypted rooms, because a client might send an unencrypted event to it.

Reply:
Right, yes. The reason I didn't is that `handleEvent` in `VerifierBase` throws and cancels when it encounters any event type that it isn't what is is expecting, like `m.room.encrypted`. I could filter out `m.room.encrypted`, but I can imagine that, especially when the verification is in the right panel (as opposed to a dialog), it will be common to have non-verification events be sent in the room while the verification is ongoing (e.g. a message, or to setup a VoIP call).

So unless there is a reason not to, instead I'd propose to just ignore events in `VerifierBase` that are non related to verification, and then indeed also listen to `event` for encrypted rooms.

Reply:
VerifierBase only receives events with the correct relation to the start event, so messages and other events unrelated to verification shouldn't cancel verification. Made the changes to ignore `m.room.encrypted` and route unencrypted events through as well.